### PR TITLE
ENH: add extraglobs option to run_doctests

### PR DESCRIPTION
**Pull request by kohr-h**
Tiny addition to doctests for the case extra modules are required, as discussed [here](https://github.com/adler-j/odl/pull/1#discussion_r136057920).

### CODE REVIEW

File: odl/util/testutils.py [hunk 1]
```diff
@@ -686,8 +693,14 @@ def run_doctests(skip_if=False):
                               'interpreter if the doctests do not work.',
                               RuntimeWarning)
 
-    testmod(optionflags=optionflags,
-            extraglobs={'odl': odl, 'np': np})
+    if extraglobs is None:
```
Comment (adler-j) [inline odl/util/testutils.py]: Weird logic here, why not:

```python
globs = {'odl': odl, 'np': np}
if extraglobs:
    globs.update(extraglobs_in)

testmod(optionflags=optionflags, extraglobs=globs)
```

Comment (kohr-h) [reply odl/util/testutils.py]: Of course 😲

**Comment by kohr-h** [comment]
> How would this work with pytest?

How wouldn't it? Apparently Travis has no issue with it, and I can't see where it should come from.

If you mean modules that have an extra module required for tests, okay, sure, that may not work (will check), but those tend to be in `contrib` anyway.
Anyhow, I'll see if there's a solution for that.

**Comment by adler-j** [comment]
Someone screwed up a rebase :D

**Comment by kohr-h** [comment]
> Someone screwed up a rebase :D

😇

**Comment by kohr-h** [comment]
Turns out that both running doctests directly and through pytest doesn't require amending the namespace. The import of `theano` or whatever in the module is sufficient (at least for me).

I'll close this and change the stuff in #1098

**Comment by adler-j** [comment]
Even if they are in contrib we even recommend in the doc:

> To test all of your code with pytest, run `pytest odl/contrib/mylib/`

**Pull request closed without merging**